docs/monitoring/autodiscovery.md

```diff
@@ -0,0 +1,60 @@
+
+<img class="content-modal-image" alt="Enabling Automatic Autodiscovery Monitoring" src="../images/monitoring/automatic-autodiscovery-monitoring.png" title="Enabling Automatic Autodiscovery Monitoring">
+
+### How to start the Autodiscovery?
+
+Simply visit Discovery option in the menu on the left side. You will be presented with instructions to install Sematext Agent. On each machine (or k8s/swarm cluster) Sematext Agent
+is installed, it will instantly start discovering services that can be monitored. Those services will be displayed in Autodiscovery table from where you can click on it to enable automatic monitoring. In most cases Sematext Agent knows how to set up the monitor without requiring any additional actions from you, which means the metrics will start appearing few seconds after you enable automatic monitoring. In other cases you may have to provide credentials so monitoring agents can connect to the service you wish to monitor (in any case, Autodiscovery UI will tell you exactly what to do).
+
+### How does Sematext Autodiscovery Monitoring work in bare-metal/virtual machine environments?
+
+It is enough to install sematext-agent rpm/deb package and set up one of your Infra App tokens by following UI instructions. This will start Sematext Agent which will
+automatically start scanning for known service types. Separately, it will connect to Sematext backend to fetch info about any automatic monitoring rules you may have defined in UI.
+
+In bare-metal/vritualMachine environments, Sematext Agent will automatically create monitoring agents only if configured so in the UI.
+
+### How does Sematext Autodiscovery Monitoring work in container environments? 
+
+The Sematext Agent container gets distributed to every Kubernetes, Docker Swarm or Docker Enterprise node. Sematext Agent installs hook points on various kernel functions via eBPF and gets notified when a process creates a socket or sends data through it. It also uses eBPF to to watch for process termination so it can shut down monitoring agent containers/pods when your service is gone. In Swarm setups specifically, Sematext Agent joins containers to its own network so it can establish communication with them.
```

Comment:
Sematext Agent installs hook points on various kernel functions via eBPF and gets notified when a process creates a socket or sends data through it. It also uses eBPF to to watch for process termination so it can shut down monitoring agent containers/pods when your service is gone. In Swarm setups specifically, Sematext Agent joins containers to its own network so it can establish communication with them. => Sematext Agent uses eBPF to watch for services starting, as well as stopping, so it can start and stop monitoring agent containers and pods.